File: php/ATM.php
<?php

do {
    $nama = readline("Masukkan Nama : ");
    $konfirmasi = readline("Apakah nama anda sudah benar? (Y/N) ");

} while ($konfirmasi == 'N' || $konfirmasi == 'n');

$nim = readline("Masukkan NIM : ");
$saldo = (int)$nim;

echo "Saldo awal anda : Rp $saldo\n";

do {
    echo "\n========== MENU ATM ==========\n";
    echo "1. Cek Saldo\n";
    echo "2. Tarik Tunai\n";
    echo "3. Setor Tunai\n";
    echo "4. Transfer\n";
    echo "5. Keluar\n";
    $menu = readline("menu ");

    switch ($menu) {

        case 1:
            echo "Saldo anda : Rp $saldo\n";
            break;

        case 2:
            $jumlah = readline("Jumlah saldo yang ingin ditarik : ");
            if ($jumlah > $saldo) {
                echo "Saldo tidak cukup!\n";
            } else {
                $saldo -= $jumlah;
                echo "berhasil. Saldo sekarang : Rp $saldo\n";
            }
            break;

        case 3:
            $jumlah = readline("Jumlah setor tunai : ");
            $saldo += $jumlah;
            echo "berhasil. Saldo sekarang : Rp $saldo\n";
            break;

        case 4:
            $bank = readline("Bank tujuan : ");
            $norek = readline("No Rekening : ");
            $jumlah = readline("Jumlah saldo yang akan ditransfer : ");
            if ($jumlah > $saldo) {
                echo "Saldo tidak cukup\n";
            } else {
                $saldo -= $jumlah;
                echo "berhasil. Saldo sekarang : Rp $saldo\n";
            }
            break;

        case 5:
            echo "Terima kasih\n";
            break;

        default:
            echo "Menu tidak valid\n";
    }

} while ($menu != 5);

?>

File: java/atm.java
import java.util.Scanner;

public class atm {
    public static void main(String[] args) {

        Scanner input = new Scanner(System.in);

        String nama;
        char konfirmasi;
        int nim, saldo, menu, jumlah;

        // Validasi Nama
        do {
            System.out.print("Masukkan Nama : ");
            nama = input.next();

            System.out.print("Apakah nama anda sudah benar? (Y/N) ");
            konfirmasi = input.next().charAt(0);

        } while (konfirmasi == 'N' || konfirmasi == 'n');


        // Input NIM -> jadi saldo
        System.out.print("Masukkan NIM : ");
        nim = input.nextInt();
        saldo = nim;

        System.out.println("Saldo awal anda : Rp " + saldo);


        // Menu ATM
        do {
            System.out.println("\n========== MENU ATM ==========");
            System.out.println("1. Cek Saldo");
            System.out.println("2. Tarik Tunai");
            System.out.println("3. Setor Tunai");
            System.out.println("4. Transfer");
            System.out.println("5. Keluar");

            System.out.print("menu ");
            menu = input.nextInt();

            switch(menu) {

                case 1:
                    System.out.println("Saldo anda : Rp " + saldo);
                    break;

                case 2:
                    System.out.print("Jumlah saldo yang ingin ditarik : ");
                    jumlah = input.nextInt();

                    if (jumlah > saldo) {
                        System.out.println("Saldo tidak cukup, tidak dapat ditarik");
                    } else {
                        saldo -= jumlah;
                        System.out.println("berhasil. Saldo sekarang : Rp " + saldo);
                    }
                    break;

                case 3:
                    System.out.print("Jumlah setor tunai : ");
                    jumlah = input.nextInt();
                    saldo += jumlah;
                    System.out.println("Setoran berhasil. Saldo sekarang : Rp " + saldo);
                    break;

                case 4:
                    System.out.print("Bank tujuan : ");
                    String bank = input.next();
                    System.out.print("No Rekening : ");
                    String norek = input.next();
                    System.out.print("Jumlah yang akan ditransfer : ");
                    jumlah = input.nextInt();

                    if (jumlah > saldo) {
                        System.out.println("Saldo tidak cukup");
                    } else {
                        saldo -= jumlah;
                        System.out.println("berhasil. Saldo sekarang : Rp " + saldo);
                    }
                    break;

                case 5:
                    System.out.println("Terima kasih");
                    break;

                default:
                    System.out.println("Menu tidak valid");
            }

        } while (menu != 5);

        input.close();
    }
}
